File: 102-binary-tree-level-order-traversal/102-binary-tree-level-order-traversal.java
class Solution {
    public List<List<Integer>> levelOrder(TreeNode root) {
        if(root == null) return new ArrayList();
        
        List<List<Integer>> res = new ArrayList();
        Queue<TreeNode> queue = new LinkedList();
        
        queue.offer(root);
        
        while(!queue.isEmpty()) {
            int size = queue.size();
            List<Integer> list = new ArrayList();
            while(size-- > 0) {
                TreeNode temp = queue.poll();
                list.add(temp.val);
                if(temp.left != null) {
                    queue.offer(temp.left);
                }
                if(temp.right != null) {
                    queue.offer(temp.right);
                }
            }
            res.add(list);
        }
        
        return res;
    }
}

File: 102-binary-tree-level-order-traversal/102-binary-tree-level-order-traversal.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def levelOrder(self, root: Optional[TreeNode]) -> List[List[int]]:
        if(root == None): return []
        
        res = []
        
        queue = deque()
        
        queue.append(root)
        
        while queue:
            size = len(queue)
            lis = []
            while size:
                temp = queue.popleft()
                lis.append(temp.val)
                if(temp.left != None):
                    queue.append(temp.left)
                if(temp.right != None):
                    queue.append(temp.right)
                size -= 1
            res.append(lis)
        
        return res